cvmfs/util/algorithm.h

```diff
@@ -107,6 +106,56 @@ class StopWatch : SingleCopy {
   timeval start_, end_;
 };
 
+/**
+ * Log2Histogram is a simple implementation of 
+ * log2 histogram data structure which stores 
+ * and prints log2 histogram. It is used for
+ * getting and printing latency metrics of 
+ * CVMFS fuse calls.
+ * 
+ * Log2Histogram hist(2);
+ * hist.Add(1);
+ * hist.Add(2);
+ * hist.PrintLog2Histogram();
+ */
+
+class Log2Histogram {
+ public:
+  explicit Log2Histogram(unsigned int nbins);
+
+  void Add(unsigned int value) {
+    unsigned int i;
+    unsigned int flag = 1;
+
+    for (i = 1; i <= this->bins_.size(); i++) {
```

Comment:
Here one can do one more small optimization:

    const unsigned N = bins_.size();
    for (i = 1; i < N; i++) {
      ...

This gives the compiler an easier time for loop optimizations